library/Icingadb/ProvidedHook/Reporting/SlaReport.php

```diff
@@ -0,0 +1,264 @@
+
+                    break;
+                case 'month':
+                    $interval = new DateInterval('P1M');
+                    $format = 'Y-m';
+                    $boundary = 'first day of next month midnight';
+
+                    break;
+            }
+
+            $dimensions = $rd->getDimensions();
+            $dimensions[] = ucfirst($config['breakdown']);
+            $rd->setDimensions($dimensions);
+
+            foreach ($this->yieldTimerange($timerange, $interval, $boundary) as list($start, $end)) {
+                foreach ($this->fetchSla(new Timerange($start, $end), $filter) as $row) {
+                    $row = $this->createReportRow($row);
+
+                    if ($row === null) {
+                        continue;
+                    }
+
+                    $dimensions = $row->getDimensions();
+                    $dimensions[] = $start->format($format);
+                    $row->setDimensions($dimensions);
+
+                    $rows[] = $row;
+                }
+            }
+        } else {
+            foreach ($this->fetchSla($timerange, $filter) as $row) {
+                $rows[] = $this->createReportRow($row);
+            }
+        }
+
+        $rd->setRows($rows);
+
+        return $rd;
+    }
+
+    /**
+     * Yield start and end times that recur at the specified interval over the given time range
+     *
+     * @param Timerange    $timerange
+     * @param DateInterval $interval
+     * @param string|null  $boundary English text datetime description for calculating bounds to get
+     *                               calendar days, weeks or months instead of relative times according to interval
+     *
+     * @return \Generator
+     */
+    protected function yieldTimerange(Timerange $timerange, DateInterval $interval, $boundary = null)
+    {
+        $start = clone $timerange->getStart();
+        $end = clone $timerange->getEnd();
+        $oneSecond = new DateInterval('PT1S');
+
+        if ($boundary !== null) {
+            $intermediate = (clone $start)->modify($boundary);
+            if ($intermediate < $end) {
+                yield [clone $start, $intermediate->sub($oneSecond)];
+
+                $start->modify($boundary);
+            }
+        }
+
+        $period = new DatePeriod($start, $interval, $end, DatePeriod::EXCLUDE_START_DATE);
+
+        foreach ($period as $date) {
+            /** @var \DateTime $date */
+            yield [$start, (clone $date)->sub($oneSecond)];
+
+            $start = $date;
+        }
+
+        yield [$start, $end];
+    }
+
+    public function initConfigForm(Form $form)
+    {
+        $form->addElement('text', 'filter', [
+            'label' => t('Filter')
+        ]);
+
+        $form->addElement('select', 'breakdown', [
+            'label'   => t('Breakdown'),
+            'options' => [
+                'none'  => 'None',
+                'day'   => 'Day',
+                'week'  => 'Week',
+                'month' => 'Month'
```

Comment:
These should be translatable. Not that for `None` you should pass a context, as otherwise it's ambiguous.